Add better coverage for block justify-self tests

davidtranhq opened the pull request:
Compared to the existing tests, these also:
* check for correct alignment on boxes with margins and padding
* check for proper stretching on replaced elements
* check that min-width is respected

And in general make the test output a bit cleaner.

davidtranhq committed:
[css] Add tests for on block-level elements

davidtranhq committed:
Add RTL case, absolute parent, and float: [left|right]

davidtranhq committed:
Test text-justify: stretch|normal

davidtranhq committed:
Add reftests for block-layout

Made changes to the previous test suite, added some more edge
cases for `justify-self: stretch`

davidtranhq committed:
Properly make table children table cells

davidtranhq committed:
Fix linter errors

davidtranhq committed:
Add stretch fallback tests

fantasai reviewed (changes requested):
Comments:

- Ensure this fits into 600px by 600px so that it works in all reftest harnesses. This might mean making the text smaller, e.g. `html { font: 10px/1 sans-serif;}`
- Nit: The backticks here are distracting. Replace with `<code>` or just remove them.
- Add some margin around your test boxes so we can figure out where they start/end. :) They're jamming together in the horizontal axis and it's a bit confusing, especially with the overflow tests.
- Nit: make it clearer what's labelling what. Maybe using `<dt>`+`<dd>` for the caption/test box? Then you can style the captions to have a little space above so it's easier to make the association. Switching to multicol would also mean you can keep them together, right now the label for the top item in column 2 is at the bottom of column 1... e.g.`.test-suite { columns: 190px; } dt { margin-top: 2px; break-after: avoid;  } dd { margin: 0 4px; border: 1px solid; }`. 
- Add some random small amounts of margins/padding/border thickness to your test boxes so we can be sure they are accounted for correctly. Also some padding to the container, same reason.
- Don't use green here, it usually means "pass" (in contrast to red). This is a "match this box to that box" test, so consider blue, teal, gray, orange, purple, whatever. I like to make the test color stand out in contrast to to the expected value, e.g. orange/gray, fuchsia/purple, etc.
- Don't link to external image files. You can use `/css/support/square-teal.png` or something else from that directory, though (or add your own file; but those should work).
- We already merged in a [block-justify-self.html](https://wpt.live/css/css-align/self-alignment/block-justify-self.html) file; if this is replacing that test, should you be removing it here?
- Suggest moving these to the /css/css-align/blocks directory to create a mini test suite about alignment in block formatting contexts. :)

davidtranhq committed:
Incorporate fixes from PR review

https://github.com/web-platform-tests/wpt/pull/47339#pullrequestreview-2208874549

davidtranhq committed:
Fix ltr direction for rtl test

davidtranhq committed:
Add better coverage for block justify-self tests

Compared to the existing tests, these also:
* check for correct alignment on boxes with margins and padding
* check for proper stretching on replaced elements
* check that min-width is respected

And in general make the test output a bit cleaner.
